lib/dentaku/ast/identifier.rb

```diff
@@ -12,6 +12,8 @@ def initialize(token)
       def value(context={})
         v = context[identifier]
```

Comment:
That is definitely cleaner, and I agree that relying on a special string 'null' as an input is strange. The problem with the way things are written at the moment is that `context` has the unbound variable's key in it whether or not it was passed in as an input, because of the way that the context hash is built up here https://github.com/rubysolo/dentaku/blob/master/lib/dentaku/bulk_expression_solver.rb#L46 ... actually looking at that now, that looks like an avoidable behavior. Be right back.